substrate/frame/revive/src/exec.rs

```diff
@@ -875,16 +915,31 @@ where
 		deposit_limit: BalanceOf<T>,
 		read_only: bool,
 		origin_is_caller: bool,
-	) -> Result<Option<(Frame<T>, E)>, ExecError> {
+	) -> Result<Option<(Frame<T>, PrecompileOrExecutable<E>)>, ExecError> {
 		let (account_id, contract_info, executable, delegate, entry_point, nested_gas) =
 			match frame_args {
 				FrameArgs::Call { dest, cached_info, delegated_call } => {
 					let contract = if let Some(contract) = cached_info {
 						contract
 					} else {
-						if let Some(contract) =
-							<ContractInfoOf<T>>::get(T::AddressMapper::to_address(&dest))
-						{
+						let addr = T::AddressMapper::to_address(&dest);
+						if is_precompile(&addr) && delegated_call.is_none() {
```

Comment:
It is legal to delegate call a precompile.

Reply:
yeah actually it is will remove this extra condition